docs/tutorialkit.dev/src/content/docs/reference/react-components.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: React components
+description: Use TutorialKit's React components for fun and profit.
+---
+import PackageManagerTabs from '@components/Tabs/PackageManagerTabs.astro'
+import { Code } from '@astrojs/starlight/components';
+import themeCSS from '@components/react-examples/theme.css?raw';
+
+import Example from '@components/react-examples/Example.astro';
+
+import ExampleSimpleEditor from '@components/react-examples/ExampleSimpleEditor';
+import ExampleFileTree from '@components/react-examples/ExampleFileTree';
+import ExampleCodeMirrorEditor from '@components/react-examples/ExampleCodeMirrorEditor';
+import ExampleTerminal from '@components/react-examples/ExampleTerminal';
+
+import sourceExampleSimpleEditor from '@components/react-examples/ExampleSimpleEditor?raw';
+import sourceExampleFileTree from '@components/react-examples/ExampleFileTree?raw';
+import sourceExampleCodeMirrorEditor from '@components/react-examples/ExampleCodeMirrorEditor?raw';
+import sourceExampleTerminal from '@components/react-examples/ExampleTerminal?raw';
+
+You can reuse TutorialKit's React components in your own applications. They are designed to work well with the [WebContainer API][webcontainers].
+
+Here's a simple editor that can be made with those components:
+
+<ExampleSimpleEditor client:load />
+
+## Installation
+
+Those components use TutorialKit's design system which is based on an atomic CSS engine, [UnoCSS](https://unocss.dev/). You will need a few dependencies in order to use them:
+
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install @tutorialkit/components-react @tutorialkit/theme unocss @iconify-json/ph fast-glob
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm install @tutorialkit/components-react @tutorialkit/theme unocss @iconify-json/ph fast-glob
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add @tutorialkit/components-react @tutorialkit/theme unocss @iconify-json/ph fast-glob
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+To setup UnoCSS with TutorialKit's components, you need to create a `uno.config.ts`:
+
+```ts title=uno.config.ts
+import { theme, rules, shortcuts } from '@tutorialkit/theme';
+import { convertPathToPattern, globSync } from 'fast-glob';
+import fs from 'node:fs/promises';
+import { join } from 'path';
+import { defineConfig, presetIcons, presetUno, transformerDirectives } from 'unocss';
+
+export default defineConfig({
+  theme,
+  rules,
+  shortcuts,
+  content: {
+    inline: globSync(
+      `${convertPathToPattern(join(require.resolve('@tutorialkit/components-react'), '..'))}/**/*.js`,
+    ).map((filePath) => {
+      return () => fs.readFile(filePath, { encoding: 'utf8' });
+    }),
+  },
+  transformers: [transformerDirectives()],
+  presets: [presetUno(), presetIcons()],
+});
+```
+
+Then you need all the CSS variables used by TutorialKit design system:
+
+<Code code={themeCSS} lang="css" title="src/theme.css" class="limit-code-height" />
```

Comment:
Nice! 😻 

Is this something that we could show on theming page as well? There it would be great to have a reference `theme.css` that could be used as base when starting to make changes in the theme.

Reply:
Added! :raised_hands: